Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs

```diff
@@ -49,6 +49,36 @@ public async Task DisposeCleansUpThreads()
     Assert.Fail("Thread count didn't drop back down after waiting 500ms.");
   }
   
+  [Test]
+  public async Task SplitOnEveryAssertion() {
+    var programString = @"
+procedure Procedure(y: int)
+  ensures y == 3;
+{
+  assert y > 2;
+  assert y > 1;
+  assert y < 4;
+}
+".Trim();
+    Parser.Parse(programString, "fakeFilename10", out var program);
+    var options = CommandLineOptions.FromArguments(TextWriter.Null);
+    options.VcsSplitOnEveryAssert = true;
+    options.PrintErrorModel = 1;
+    var engine = ExecutionEngine.CreateWithoutSharedCache(options);
+    var tasks = engine.GetVerificationTasks(program);
+    
+    // The first split is empty. Maybe it can be optimized away
```

Comment:
Yeah, I've been wondering about the empty first split for a while now. It'd be nice to get rid of it.